ush/forecast_predet.sh

```diff
@@ -0,0 +1,299 @@
+
+	# Model specific stuff
+	FCSTEXECDIR=${FCSTEXECDIR:-$HOMEgfs/sorc/fv3gfs.fd/NEMS/exe}
+	FCSTEXEC=${FCSTEXEC:-fv3_gfs.x}
+	PARM_FV3DIAG=${PARM_FV3DIAG:-$HOMEgfs/parm/parm_fv3diag}
+	PARM_POST=${PARM_POST:-$HOMEgfs/parm/post}
+
+        # Wave coupling parameter defaults to false
+        cplwav=${cplwav:-.false.}
+
+	# Model config options
+	APRUN_FV3=${APRUN_FV3:-${APRUN_FCST:-${APRUN:-""}}}
+	#the following NTHREAD_FV3 line is commented out because NTHREAD_FCST is not defined 
+	#and because NTHREADS_FV3 gets overwritten by what is in the env/${macine}.env
+	#file and the value of npe_node_fcst is not correctly defined when using more than
+	#one thread and sets NTHREADS_FV3=1 even when the number of threads is appropraitely >1
+	#NTHREADS_FV3=${NTHREADS_FV3:-${NTHREADS_FCST:-${nth_fv3:-1}}}
+	NTHREADS_FV3=${nth_fv3:-1} 
+	cores_per_node=${cores_per_node:-${npe_node_fcst:-24}}
+	ntiles=${ntiles:-6}
+	if [ $MEMBER -lt 0 ]; then
+		NTASKS_TOT=${NTASKS_TOT:-$npe_fcst_gfs}
+	else
+		NTASKS_TOT=${NTASKS_TOT:-$npe_efcs}
+	fi
+
+	TYPE=${TYPE:-"nh"}                  # choices:  nh, hydro
+	MONO=${MONO:-"non-mono"}            # choices:  mono, non-mono
+
+	QUILTING=${QUILTING:-".true."}
+	OUTPUT_GRID=${OUTPUT_GRID:-"gaussian_grid"}
+	OUTPUT_FILE=${OUTPUT_FILE:-"nemsio"}
+	WRITE_NEMSIOFLIP=${WRITE_NEMSIOFLIP:-".true."}
+	WRITE_FSYNCFLAG=${WRITE_FSYNCFLAG:-".true."}
+        affix="nemsio"
+        [[ "$OUTPUT_FILE" = "netcdf" ]] && affix="nc"
+
+	rCDUMP=${rCDUMP:-$CDUMP}
+
+	#------------------------------------------------------------------
+	# setup the runtime environment
+	if [ $machine = "WCOSS_C" ] ; then
+		HUGEPAGES=${HUGEPAGES:-hugepages4M}
+		. $MODULESHOME/init/sh 2>/dev/null
+		module load iobuf craype-$HUGEPAGES 2>/dev/null
+		export MPICH_GNI_COLL_OPT_OFF=${MPICH_GNI_COLL_OPT_OFF:-MPI_Alltoallv}
+		export MKL_CBWR=AVX2
+		export WRTIOBUF=${WRTIOBUF:-"4M"}
+		export NC_BLKSZ=${NC_BLKSZ:-"4M"}
+		export IOBUF_PARAMS="*nemsio:verbose:size=${WRTIOBUF},*:verbose:size=${NC_BLKSZ}"
+	fi
+
+	#-------------------------------------------------------
+        if [ ! -d $ROTDIR ]; then mkdir -p $ROTDIR; fi
+        mkdata=NO
+        if [ ! -d $DATA ]; then
+                mkdata=YES
+                mkdir -p $DATA ;
+        fi
+        cd $DATA || exit 8
+        mkdir -p $DATA/INPUT
+
+	#------------------------------------------------------------------
+	# changeable parameters
+	# dycore definitions
+	res=$(echo $CASE |cut -c2-5)
+	resp=$((res+1))
+	npx=$resp
+	npy=$resp
+	npz=$((LEVS-1))
+	io_layout="1,1"
+	#ncols=$(( (${npx}-1)*(${npy}-1)*3/2 ))
+
+	# spectral truncation and regular grid resolution based on FV3 resolution
+	JCAP_CASE=$((2*res-2))
+	LONB_CASE=$((4*res))
+	LATB_CASE=$((2*res))
+	if [ $LATB_CASE -eq 192 ]; then
+		LATB_CASE=190 # berror file is at this resolution
+	fi
+
+	JCAP=${JCAP:-$JCAP_CASE}
+	LONB=${LONB:-$LONB_CASE}
+	LATB=${LATB:-$LATB_CASE}
+
+	LONB_IMO=${LONB_IMO:-$LONB_CASE}
+	LATB_JMO=${LATB_JMO:-$LATB_CASE}
```

Comment:
where in the forecast model is this used?

Reply:
JCAP/LONB/LATB are still used in some of the source fix file names.

Reply:
forgot about those.  Thanks!